game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1428,98 +1429,22 @@ public static void triggerPlayerPropertyChange(final Set<TriggerAttachment> sati
-              }
-              if (clearFirst && newValue.length() < 1) {
-                change.add(ChangeFactory.attachmentPropertyReset(attachment, property.getFirst()));
-              } else {
-                change.add(
-                    ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-              }
-              bridge.getHistoryWriter()
-                  .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                      + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                      + t.getPlayerAttachmentName().getSecond() + " attached to " + player.getName());
-              break;
-            }
-            case "PoliticalActionAttachment": {
-              final PoliticalActionAttachment attachment =
-                  PoliticalActionAttachment.get(player, t.getPlayerAttachmentName().getSecond());
-              if (newValue.equals(attachment.getRawPropertyString(property.getFirst()))) {
-                continue;
-              }
-              if (clearFirst && newValue.length() < 1) {
-                change.add(ChangeFactory.attachmentPropertyReset(attachment, property.getFirst()));
-              } else {
-                change.add(
-                    ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-              }
-              bridge.getHistoryWriter()
-                  .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                      + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                      + t.getPlayerAttachmentName().getSecond() + " attached to " + player.getName());
-              break;
-            }
-            case "UserActionAttachment": {
-              final UserActionAttachment attachment =
-                  UserActionAttachment.get(player, t.getPlayerAttachmentName().getSecond());
+          // TODO: Change to use 'Map.of()' factory method once using Java 9+.
+          // TODO add other attachment changes here if they attach to a player
+          final Map<String, BiFunction<PlayerId, String, DefaultAttachment>> attachmentNameToAttachmentGetter =
+              new HashMap<>();
+          attachmentNameToAttachmentGetter.put("PlayerAttachment", PlayerAttachment::get);
+          attachmentNameToAttachmentGetter.put("RulesAttachment", RulesAttachment::get);
+          attachmentNameToAttachmentGetter.put("TriggerAttachment", TriggerAttachment::get);
+          attachmentNameToAttachmentGetter.put("TechAttachment", TechAttachment::get);
+          attachmentNameToAttachmentGetter.put("PoliticalActionAttachment", PoliticalActionAttachment::get);
+          attachmentNameToAttachmentGetter.put("UserActionAttachment", UserActionAttachment::get);
+          final String attachmentName = t.getPlayerAttachmentName().getFirst();
+          {
```

Comment:
Is there a specific reason to enclose the following code in a separate block?
Now that the code is no longer duplicated this should no longer be necessary.

Reply:
Good point, I mostly did that such that the 'diff' would look nicer :smiley: . I have fixed it in the most recent commit.